### ISSUE TITLE
Add `transitionToTop` and others

### ISSUE THREAD

**Issue by grabbou** [exponentjs/ex-navigator#7]
Hey, thanks for the module! 

Since it's somehow similar to my in-house navigator, I wanted to suggest adding a quite a few custom methods I often find useful so I can switch to this library and keep it maintaining with you :)

First one is `transitionToTop` (I use `transitionTo` instead of `push` hence that name)

``` js
function transitionToTop(route) {
    this.navigator.replaceAtIndex(route, 0);
    this.navigator.popToTop();
}
```

It's quite useful when you want to transition to a new route (push it), but you want to make sure you clear your route stack. This is extremely useful when you have auth flow with multiple routes and on successful login you want to navigate to protected home screen making sure none of auth routes are left in stack (memory management for the win)

Another one is `popBack(index)` to pop back n times. Say you open help section, then you allow your users to select a subject of a message and then you push a new route with a contact form after subject is selected. On successful form submission, you want to make sure your user goes back to `help` screen, not to `select title of a message` one.

``` js
function popBack(index) {
    const routes = this.navigator.getCurrentRoutes();
    this.navigator.popToRoute(routes[routes.length - index - 1]);
}
```

**Comment by ide** [comment]
Thanks for the suggestions! I've been meaning to move [everything below 139 here](https://github.com/exponentjs/ex-navigator/blob/master/ExNavigator.js#L139) into a mixin. The mixin would also be a good place to add your suggested methods. It's late here in my time zone so it'll be awhile.

**Comment by grabbou** [comment]
Sounds good and pretty reasonable! If you need any help with that, do let me know

**Comment by ide** [comment]
Published 0.2.3 to npm with the new methods: https://github.com/exponentjs/ex-navigator/blob/master/ExNavigatorMixin.js#L55

**Comment by grabbou** [comment]
Thank you! :)
